src/lib/translators/mongo_dates.ts

```diff
@@ -180,6 +175,58 @@ export const ADVANCED_DATE_EXTRACT_MAP: Record<
   }),
 };
 
+export const ADVANCED_DATE_EXTRACT_MAP_MONGO_5: Record<
+  S.AdvancedDateInfo,
+  (step: Readonly<S.DateExtractStep>) => object
+> = {
+  ...ADVANCED_DATE_EXTRACT_MAP,
+  firstDayOfWeek: step => ({
+    // [ ] TO FIX using Mongo 5
+    // We subtract to the target date a number of days corresponding to (dayOfWeek - 1)
+    $subtract: [
+      $$(step.column),
+      {
+        $multiply: [{ $subtract: [{ $dayOfWeek: $$(step.column) }, 1] }, 24 * 60 * 60 * 1000],
+      },
+    ],
+  }),
```

Comment:
For Mongo 5, the easiest is to use the $dateTrunc operator, which does exactly what you need with the unit `week`.
https://docs.mongodb.com/manual/reference/operator/aggregation/dateTrunc/
Beware that `startOfWeek` defaults to sunday, so we may want to precise monday for our use case.

And actually, the same operator can be leverage for other extractions in mongo 5+. Maybe it's worth using it not only for weeks but for first day of month, first day of quarter, etc.